includes/class-wc-stripe-helper.php

```diff
@@ -419,29 +421,51 @@ public static function get_legacy_payment_method( $id ) {
 	 */
 	public static function get_legacy_available_payment_method_ids() {
 		$stripe_settings            = get_option( 'woocommerce_stripe_settings', [] );
+		$payment_method_classes     = self::get_legacy_payment_method_classes();
 		$ordered_payment_method_ids = isset( $stripe_settings['stripe_legacy_method_order'] ) ? $stripe_settings['stripe_legacy_method_order'] : [];
 
 		// If the legacy method order is not set, return the default order.
 		if ( ! empty( $ordered_payment_method_ids ) ) {
 			$payment_method_ids = array_map(
-				function( $payment_method_class ) {
-					if ( 'stripe' === $payment_method_class ) {
+				function( $payment_method_id ) {
+					if ( 'stripe' === $payment_method_id ) {
 						return 'card';
 					} else {
-						return str_replace( 'stripe_', '', $payment_method_class );
+						return str_replace( 'stripe_', '', $payment_method_id );
 					}
 				},
 				$ordered_payment_method_ids
 			);
+
+			// Cover the edge case when new Stripe payment methods are added to the plugin which do not exist in
+			// the `stripe_legacy_method_order` option.
+			if ( count( $payment_method_ids ) - 1 !== count( $payment_method_classes ) ) {
+				$remaining_payment_method_ids = array_reduce(
+					$payment_method_classes,
+					function( $ids, $payment_method_class ) use ( $payment_method_ids ) {
+						$id = str_replace( 'stripe_', '', $payment_method_class::ID );
+						if ( ! in_array( $id, $payment_method_ids, true ) ) {
+							$ids[] = $id;
+						}
+
+						return $ids;
+					}
+				);
+
+				$payment_method_ids = array_merge( $payment_method_ids, $remaining_payment_method_ids );
```

Comment:
At first this was a little tough to wrap my head around. I think I understand it now but I wonder if using a loop would make this easier to follow.

```suggestion
				foreach ( $payment_method_classes as $payment_method_class ) {
					$id = str_replace('stripe_', '', $payment_method_class::ID);

					if ( ! in_array( $id, $payment_method_ids, true ) ) {
						$payment_method_ids[] = $id;
					}
				}
```

Note you can get rid of the array merge because the loop approach is just appending each missing element to the end of the original array anyway.